### PR TITLE
mysql(ticdc): remove error filter when check isTiDB in backend init

### LINKED ISSUE #11213: ticdc: Removed error type detection in `CheckIsTiDB`
Different downstreams will have different error messages for "select tidb_version()". We cannot provide a complete error whitelist in `CheckIsTiDB`. 
Considering that we just want to know whether the downstream is tidb, the specific error cause is not very important. Therefore, we can directly output error information for all errors instead of returning errors.

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: close #11213

### What is changed and how it works?

remove error detection in `checkIsTiDB` to avoid unexcept error when the downstream is different mysql-class server

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by asddongmen — APPROVED** [review]

File: pkg/sink/mysql/db_helper.go [hunk 1]
```diff
@@ -303,11 +303,6 @@ func CheckIsTiDB(ctx context.Context, db *sql.DB) (bool, error) {
 	err := row.Scan(&tidbVer)
 	if err != nil {
 		log.Error("check tidb version error", zap.Error(err))
-		// downstream is not TiDB, do nothing
-		if mysqlErr, ok := errors.Cause(err).(*dmysql.MySQLError); ok && (mysqlErr.Number == tmysql.ErrNoDB ||
-			mysqlErr.Number == tmysql.ErrSpDoesNotExist || mysqlErr.Number == tmysql.ErrDBaccessDenied) {
-			return false, nil
-		}
 		return false, errors.Trace(err)
```
Comment (asddongmen) [inline pkg/sink/mysql/db_helper.go:306]:
```suggestion
		return false, nil
```

**Comment by asddongmen** [comment]
Please refine this PR description and release note.

**Review by CharlesCheung96 — APPROVED** [review]

**Comment by lidezhu** [comment]
If `select tidb_version()` return a network error when the downstream is tidb, it may think the downstream is mysql which is a wrong assumption. And a wrong assumption may cause some potential confusing problem later.
But the chances of this happening are very small, so it is ok to fix it in this way.

**Pull request merged**